app/models/audit/display_item.rb

```diff
@@ -0,0 +1,129 @@
+
+    begin
+      klass = version.item_type.constantize
+      @entity_name = compute_entity_name(klass)
+      @entity_display_name = compute_entity_display_name(klass)
+    rescue NameError
+      @error = true
+      @entity_name = version.item_type
+      @entity_display_name = version.item_type
+    end
+
+    begin
+      changeset = version.changes_with_computed_fallback unless @error
+    rescue StandardError
+      @error = true
+    end
+
+    @changes = if @error
+      ['Error loading changes']
+    elsif klass.respond_to?(:describe_changes)
+      klass.describe_changes(version, changeset, @excluded_fields)
+    else
+      # Fallback for models without describe_changes method
+      describe_changes_fallback(version, changeset)
+    end
+  end
+
+  protected
+
+  def compute_username(users_by_id, user_id)
+    return if version.anonymous?
+    return unless user_id
+
+    user = users_by_id[user_id.to_i]
+    user&.name.presence || "User ID #{user_id}"
+  end
+
+  def compute_entity_name(klass)
+    # Special handling for GroupViewableEntity - show the referenced entity type
+    if klass == GrdaWarehouse::GroupViewableEntity
+      begin
+        record = klass.with_deleted.find_by(id: version.item_id)
+        if record&.entity_type
+          record.entity_type
+        else
+          version.item_type
+        end
+      rescue StandardError
+        version.item_type
+      end
+    else
+      klass.name
+    end
+  end
+
+  def compute_entity_display_name(klass)
+    # Try to get the actual record and use its name method first
+    if klass
+      begin
+        record = klass.with_deleted.find_by(id: version.item_id)
+        if record.respond_to?(:entity_name)
+          return record.entity_name
+        elsif record.respond_to?(:name)
+          return record.name
+        end
+      rescue StandardError
+        # If we can't load the record, fall back to humanized entity type
+      end
+    end
+
+    # Fallback to humanized entity type
+    version.item_type.underscore.humanize
+  end
+
+  def describe_changes_fallback(version, changeset)
+    entity_display_name = compute_entity_display_name(nil)
+
+    case version.event
+    when 'create'
+      ["Created #{entity_display_name}"]
+    when 'update'
+      if changeset.present?
+        changeset.map do |field, values|
+          from, to = values
+          "Changed #{field.humanize.titleize}: from #{render_changed_value(field, from)} to #{render_changed_value(field, to)}"
+        end
+      else
+        ["Modified #{entity_display_name}"]
+      end
+    when 'destroy'
+      ["Deleted #{entity_display_name}"]
+    else
+      ["Modified #{entity_display_name}"]
+    end
+  end
+
+  def render_changed_value(_field, value)
+    return 'nil' if value.nil?
+    return 'true' if value == true
+    return 'false' if value == false
+    return value.to_s if value.is_a?(Numeric)
+    return value.to_s if value.is_a?(String)
```

Comment:
Maybe?
```suggestion
    return value.to_s if value == true || value == false || value.is_a?(Numeric) || value.is_a?(String) 
```

Reply:
Actually, I think you can just drop all lines after `return 'nil' if value.nil?` (and the `return` from the last line) as the default is to call `value.to_s`